### DISCUSSION TITLE
What's the minimum GPU requirement for finetuning whisper for the low resource languages?

### DISCUSSION THREAD

**Discussion by bnriiitb** [openai/whisper#599] [Q&A]
I have ~1000 hrs of labelled data for one of the low-resource languages. I want to fine-tune the Whisper on the same. Please help me with the minimum GPU requirement, and the approximate time it would take to fine-tune.

PS: High end GPU is always good to speed up the fine-tuning; however, knowing the minimum config and fine-tuning time helps me estimate compute cost.

**Comment by wwbnjsace** [comment]
> I have ~1000 hrs of labelled data for one of the low-resource languages. I want to fine-tune the Whisper on the same. Please help me with the minimum GPU requirement, and the approximate time it would take to fine-tune.
> 
> PS: High end GPU is always good to speed up the fine-tuning; however, knowing the minimum config and fine-tuning time helps me estimate compute cost.

similarly，i want to know how to fine tune the whisper in our data，can you tell me how to fine tune thewhisper。any code exist？

**Reply by bnriiitb** [reply]
Check this notebook https://github.com/bnriiitb/whisper-fine-tuning/blob/main/fine_tune_whisper.ipynb

**Comment by farbodbj** [comment]
> I have ~1000 hrs of labelled data for one of the low-resource languages. I want to fine-tune the Whisper on the same. Please help me with the minimum GPU requirement, and the approximate time it would take to fine-tune.
> 
> PS: High end GPU is always good to speed up the fine-tuning; however, knowing the minimum config and fine-tuning time helps me estimate compute cost.

These two papers [1](https://link.springer.com/content/pdf/10.1186/s13636-024-00349-3.pdf) and [2](https://arxiv.org/html/2412.15726v1) on fine-tuning whisper has mentioned the following details about their training hardware:

1- 
For fine-tunring whisper large-v2 using AdamW optimizer:
> We used a single 48GB NVIDIA A40 GPU, the batch size for all languages was 8.


2- 
> Using gradient check-pointing [[25](https://arxiv.org/html/2412.15726v1#bib.bib25)], gradient accumulation, and an 8-bit optimizer [[26](https://arxiv.org/html/2412.15726v1#bib.bib26)], we achieve an effective batch size of 256 on a single NVIDIA A100 40 GB GPU. Gradient check-pointing is applied to both the encoder and decoder, with 16 gradient accumulation steps and a per-step batch size of 16. Additionally, we apply stochastic depth [[27](https://arxiv.org/html/2412.15726v1#bib.bib27)] to the encoder and decoder blocks. This setup results in a mixed-precision training run that takes about 42 hours.

P.S: I know 3 years have passed since the question was asked but I hope this can someone else in search of this info :)
